pkg/rancher/migrations.go

```diff
@@ -239,26 +241,32 @@ func addWebhookConfigToCAPICRDs(crdClient controllerapiextv1.CustomResourceDefin
 		return err
 	}
 	for _, crd := range crds.Items {
-		if crd.Spec.Group != "cluster.x-k8s.io" || (crd.Spec.Conversion != nil && crd.Spec.Conversion.Strategy != apiextv1.NoneConverter) {
-			continue
-		}
+		if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			latestCrd, err := crdClient.Get(crd.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if latestCrd.Spec.Group != "cluster.x-k8s.io" || (latestCrd.Spec.Conversion != nil && latestCrd.Spec.Conversion.Strategy != apiextv1.NoneConverter) {
```

Comment:
Let's move this outside the retry because this won't change and we won't `Get` if we don't need to.

Reply:
Moved